overlord/snapstate/check_snap.go

```diff
@@ -369,8 +369,42 @@ func checkBases(st *state.State, snapInfo, curInfo *snap.Info, flags Flags) erro
 	return fmt.Errorf("cannot find required base %q", snapInfo.Base)
 }
 
+func checkEpochs(_ *state.State, snapInfo, curInfo *snap.Info, _ Flags) error {
+	if curInfo == nil {
+		return nil
+	}
+	if snapInfo.Epoch.CanRead(curInfo.Epoch) {
+		return nil
+	}
+	if snapInfo.SideInfo.Revision.Store() {
+		return fmt.Errorf("cannot refresh snap %q as new revision (%s) has epoch %s that can't read current revision's epoch of %s", snapInfo.InstanceName(), snapInfo.SideInfo.Revision, snapInfo.Epoch, curInfo.Epoch)
```

Comment:
How about this one: `cannont refresh snap %q, new revision (%s, epoch %s) is incompatible with current (%s, epoch %s)`

Reply:
is this error case tested btw?